packages/core/src/physics/DynamicCollider.ts

```diff
@@ -103,14 +105,40 @@ export class DynamicCollider extends Collider {
   set mass(value: number) {
     if (this._mass !== value) {
       this._mass = value;
-      (<IDynamicCollider>this._nativeCollider).setMass(value);
+      if (this._automaticInertiaTensor || this._automaticCenterOfMass) {
+        this._setMassAndUpdateInertia();
+      } else {
+        (<IDynamicCollider>this._nativeCollider).setMass(value);
+      }
+    }
+  }
+
+  /**
+   * Whether or not to calculate the center of mass automatically, if true, the center of mass will be calculated based on the shapes added to the collider.
+   */
```

Comment:
I think is not only `on the shapes added to the collider`?

Reply:
opted